pallets/asset/src/lib.rs

```diff
@@ -1616,11 +1624,30 @@ impl<T: Trait> Module<T> {
             .checked_add(&value)
             .ok_or(Error::<T>::BalanceOverflow)?;
 
-        let ops: Vec<_> = iter::repeat(ProtocolOp::AssetCheckpoint).take(2).collect();
+        let is_checkpoint_due = |did| Self::is_checkpoint_due(ticker, did);
+        let maybe_due_checkpoint_from = is_checkpoint_due(from_portfolio.did);
+        let maybe_due_checkpoint_to = is_checkpoint_due(to_portfolio.did);
+        let num_ops = vec![&maybe_due_checkpoint_from, &maybe_due_checkpoint_to]
+            .iter()
+            .filter(|m| m.is_some())
+            .count();
+        let ops: Vec<_> = iter::repeat(ProtocolOp::AssetCheckpoint)
+            .take(num_ops)
+            .collect();
         T::ProtocolFee::charge_fees(ops.as_ref())?;
```

Comment:
```suggestion
        let ops = [&maybe_due_checkpoint_from, &maybe_due_checkpoint_to]
            .iter()
            .filter_map(core::convert::identity)
            .map(|_| ProtocolOp::AssetCheckpoint)
            .collect::<arrayvec::ArrayVec::<[_; 2]>>();
        T::ProtocolFee::charge_fees(&*ops)?;
```

Reply:
Thanks for the illuminating options! Unfortunately we only need at most one fee. This is because there is exactly one asset involved in the transaction. This is an example when (ticker, did) was misleading as a storage key for checkpoints.